On devel-2 TestCobraFluxAnalysis.test_single_gene_deletion_moma fails

Same story as in #407, the test fails locally.

```python
self = <cobra.test.test_flux_analysis.TestCobraFluxAnalysis object at 0x7f54c2ac5f98>
model = <Model e_coli_core at 0x7f54c2ac5390>

    def test_single_gene_deletion_moma(self, model):
        try:
            sutil.get_solver_name(qp=True)
        except sutil.SolverNotFound:
            pytest.skip("no qp support")

        # expected knockouts for textbook model
        growth_dict = {"b0008": 0.87, "b0114": 0.71, "b0116": 0.56,
                       "b2276": 0.11, "b1779": 0.00}

        rates, statuses = single_gene_deletion(model,
                                               gene_list=growth_dict.keys(),
>                                              method="moma")

cobra/test/test_flux_analysis.py:161:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
cobra/flux_analysis/single_deletion.py:231: in single_gene_deletion
    solver=solver, **solver_args)
cobra/flux_analysis/single_deletion.py:342: in single_gene_deletion_moma
    moma.add_moma(m)
cobra/flux_analysis/moma.py:71: in add_moma
    model.objective = prob.Objective(new_obj, direction='min')
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <optlang.gurobi_interface.Objective object at 0x7f54c28deda0>
expression = moma_dist_ACALD**2 + moma_dist_ACALDt**2 + moma_dist_ACKr**2 + moma_dist_ACONTa**2 + moma_dist_ACONTb**2 + moma_dist_A... moma_dist_SUCOAS**2 + moma_dist_TALA**2 + moma_dist_THD2**2 + moma_dist_TKT1**2 + moma_dist_TKT2**2 + moma_dist_TPI**2
sloppy = False, args = (), kwargs = {'direction': 'min'}

    def __init__(self, expression, sloppy=False, *args, **kwargs):
        super(Objective, self).__init__(expression, *args, **kwargs)
        self._expression_expired = False
        if not (sloppy or self.is_Linear):  # or self.is_Quadratic: # QP is not yet supported
>           raise ValueError("The given objective is invalid. Must be linear or quadratic (not yet supported")
E           ValueError: The given objective is invalid. Must be linear or quadratic (not yet supported

../../.virtualenvs/cobra/lib/python3.5/site-packages/optlang/gurobi_interface.py:318: ValueError
_
```

--- a/cobra/test/test_solver_utils.py
+++ b/cobra/test/test_solver_utils.py
@@ -31,8 +31,9 @@ class TestHelpers:
         assert su.choose_solver(model, "cglpk")[0]
 
         if any(s in su.solvers for s in su.qp_solvers):
-            qp_choice = su.interface_to_str(su.choose_solver(model, qp=True))
-            assert qp_choice in su.qp_solvers
+            legacy, qp_choice = su.choose_solver(model, qp=True)
+            assert not legacy
+            assert su.interface_to_str(qp_choice) in su.qp_solvers
         else:
             with pytest.raises(su.SolverNotFound):
                 su.choose_solver(model, qp=True)

--- a/cobra/util/solver.py
+++ b/cobra/util/solver.py
@@ -179,10 +179,10 @@ def get_solver_name(mip=False, qp=False):
     """
     if len(solvers) == 0:
         raise SolverNotFound("no solvers installed")
-    # glpk only does lp, not qp. Gurobi and cplex are better at mip
-    mip_order = ["gurobi", "cplex", "mosek", "glpk"]
-    lp_order = ["glpk", "cplex", "gurobi", "mosek", "scipy"]
-    qp_order = ["gurobi", "cplex", "mosek"]
+    # Those lists need to be update as optlang implements more solvers
+    mip_order = ["gurobi", "cplex", "glpk"]
+    lp_order = ["glpk", "cplex", "gurobi"]
+    qp_order = ["cplex"]
 
     if mip is False and qp is False:
         for solver_name in lp_order:

--- a/cobra/util/solver.py
+++ b/cobra/util/solver.py
@@ -179,7 +179,7 @@ def get_solver_name(mip=False, qp=False):
     """
     if len(solvers) == 0:
         raise SolverNotFound("no solvers installed")
-    # Those lists need to be update as optlang implements more solvers
+    # Those lists need to be updated as optlang implements more solvers
     mip_order = ["gurobi", "cplex", "glpk"]
     lp_order = ["glpk", "cplex", "gurobi"]
     qp_order = ["cplex"]

